kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/ProxyReconciler.java

```diff
@@ -193,10 +270,74 @@ private static String conditionMessage(@Nullable Exception exception) {
 
     @NonNull
     private static String conditionReason(@Nullable Exception exception) {
-        return exception == null ? "" : exception.getClass().getSimpleName();
+        if (exception == null) {
+            return "";
+        }
+        else {
+            return exception.getClass().getSimpleName();
+        }
     }
 
     private static boolean isReadyEqualsTrue(Conditions oldReady) {
         return Conditions.Status.TRUE.equals(oldReady.getStatus());
     }
+
+    @Override
+    public Map<String, EventSource> prepareEventSources(EventSourceContext<KafkaProxy> context) {
+        var eventSources = new ArrayList<InformerEventSource<GenericKubernetesResource, KafkaProxy>>(this.runtimeDecl.filterApis().size());
+        for (var filterKind : this.runtimeDecl.filterApis()) {
+            try {
+                eventSources.add(eventSourceForFilter(context, filterKind));
+            }
+            catch (Exception e) {
+                throw new OperatorConfigurationException("EventSource for " + filterKind + " could not be created.\n"
+                        + "Hints:\n"
+                        + "1. Check the Kind '" + filterKind.kind() + "' is present in the output of "
+                        + "`kubectl api-resources --api-group=" + filterKind.group() + "`.\n"
+                        + "2. Check access controls allow the operator to 'get,list,watch' this API.",
+                        e);
+            }
+        }
+        return EventSourceInitializer.nameEventSources(eventSources.toArray(new EventSource[0]));
+    }
+
+    @NonNull
+    private static InformerEventSource<GenericKubernetesResource, KafkaProxy> eventSourceForFilter(
+                                                                                                   EventSourceContext<KafkaProxy> context,
+                                                                                                   FilterApiDecl filterApiDecl) {
+
+        SecondaryToPrimaryMapper<GenericKubernetesResource> filterToProxy = (GenericKubernetesResource filter) -> {
+            // filters don't point to a proxy, but must be in the same namespace as the proxy/proxies which reference the,
+            // so when a filter changes we reconcile all the proxies in the same namespace
```

Comment:
If we did Gateway API I think the more natural model is for the `KafkaRoute` and the `filters` it refers to to live in the same namespace, which might be distinct from the ns of the `Gateway`(s) those `KafkaRoute`(s) refer to.

Reply:
> I'm comfortable with this for now, but do we want to maintain this restriction longer term?

There have been several RFEs against Kubernetes requesting cross namespace references between objects (secrets is the one I remember several times).  I believe they have been mostly rejected (I think an exception was made within the Gateway API itself).   I imagine it could lead to awkward finaliser corner cases.